src/wallet/wallet.h

```diff
@@ -1544,5 +1542,162 @@ class AddSpendingKeyToWallet : public boost::static_visitor<KeyAddResult>
     KeyAddResult operator()(const libzcash::InvalidEncoding& no) const;
 };
 
+struct LogTx
+{
+    std::map<std::string, CAmount> balances;
+    enum TransactionStateType { TX_GENERATED, TX_MINED };
+    std::map<uint256, TransactionStateType> txstate;
+
+    std::string ValueFromAmount(const CAmount& amount)
+    {
+        bool sign = amount < 0;
+        int64_t n_abs = (sign ? -amount : amount);
+        int64_t quotient = n_abs / COIN;
+        int64_t remainder = n_abs % COIN;
+        return strprintf("%s%d.%08d", sign ? "-" : "", quotient, remainder);
+    }
+
+    void LogBalance(const std::string& address, const CAmount& amount, const bool sum = false)
+    {
+        if (balances.count(address) != 0) {
+            if (balances.at(address) != amount && amount != 0) {
```

Comment:
If `sum` is false, it looks like the `amount != 0` condition would cause a transaction that fully-spends the funds in an address to not be logged. I assume this condition is present to prevent logging transactions that don't change the balance of an address when `sum` is true.